File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s527301188_commented.cpp
#include <iostream>
#include <sstream>
#include <vector>

int main() {
    std::string input;
    std::getline(std::cin, input);
    std::istringstream iss(input);
    std::vector<int> nums;
    int top = 0;

    std::string token;
    while (iss >> token) {
        if (token == "+" || token == "-" || token == "*") {
            int a = nums[top - 2];
            int b = nums[top - 1];
            if (token == "+") {
                nums[top - 2] = a + b;
            } else if (token == "-") {
                nums[top - 2] = a - b;
            } else if (token == "*") {
                nums[top - 2] = a * b;
            }
            top -= 1;
        } else {
            nums.push_back(std::stoi(token));
            top++;
        }
    }
    std::cout << nums[0] << std::endl;
    // End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s527301188_commented.py
import sys

input_str = input().split()
nums = [0] * len(input_str)
top = 0

for token in input_str:
    if token[0] == '+':
        top -= 1
        nums[top] = nums[top] + nums[top + 1]
    elif token[0] == '-':
        top -= 1
        nums[top] = nums[top] - nums[top + 1]
    elif token[0] == '*':
        top -= 1
        nums[top] = nums[top] * nums[top + 1]
    else:
        nums[top] = int(token)
        top += 1

print(nums[0])  # End of Code
